cpp/include/raft/matrix/select_k.cuh

```diff
@@ -76,6 +77,8 @@ namespace raft::matrix {
  *   whether to select k smallest (true) or largest (false) keys.
  * @param[in] sorted
  *   whether to make sure selected pairs are sorted by value
+ * @param[in] algo
+ *   the selection algorithm to use
```

Comment:
Can we expand the algorithms here a bit to give a brief summary of what they mean? I expect this API to be used mostly by power users, but I can't imagine even power users (outside of RAFT immediate developers) would be able to use these options without a brief description and summary of why each might be used. It's also a great opportunity to add the "see also" for the air-top-k paper.